src/test/java/hudson/plugins/git/GitSCMTest.java

```diff
@@ -2286,7 +2288,29 @@ public void testBuildEnvVarsLocalBranchNotSet() throws Exception {
        verify(buildData, times(1)).hasBeenReferenced(anyString());
        verify(build, times(1)).getActions(BuildData.class);
     }
-    
+
+    @Issue("JENKINS-38241")
+    @Test
+    public void testCommitMessageIsPrintedToLogs() throws Exception {
+        sampleRepo.init();
+        sampleRepo.write("file", "v1");
+        sampleRepo.git("commit", "--all", "--message=test commit");
+        FreeStyleProject p = setupSimpleProject("master");
+        Run<?,?> run = rule.buildAndAssertSuccess(p);
+        TaskListener mockListener = Mockito.mock(TaskListener.class);
+        Mockito.when(mockListener.getLogger()).thenReturn(Mockito.spy(StreamTaskListener.fromStdout().getLogger()));
+
+        p.getScm().checkout(run, new Launcher.LocalLauncher(listener),
+                new FilePath(run.getRootDir()).child("tmp-" + "master"),
+                mockListener, null, SCMRevisionState.NONE);
+
+        ArgumentCaptor<String> logCaptor = ArgumentCaptor.forClass(String.class);
+        verify(mockListener.getLogger(), atLeastOnce()).println(logCaptor.capture());
+        List<String> values = logCaptor.getAllValues();
+        assertTrue("checkout command should print commit message",
```

Comment:
As a matter of preference, I'd prefer a Hamcrest assertion here to make the diagnosis easier if the assertion ever fails.